dial_test.go

```diff
@@ -485,9 +469,7 @@ func testDialAccept(opts testDialOpts, l net.Listener) chan dialServerActual {
 
 		// Read Auth request.
 		authRequestExpected := authRequestExpectedChapSha1
-		if opts.isPapSha256Auth {
-			authRequestExpected = authRequestExpectedPapSha256
```

Comment:
Please, add:

* Unit tests with `PapSha256Auth` for `AuthDialer`.
* Integration test with Tarantool CE and NetDialer + Greeting + Protocol + Auth with `PapSha256Auth` (Tarantool should return an error).

Reply:
`NetDialer` does not have the `Auth` field, so user cannot choose the auth method for this dialer. It is set to always be `ChapSha1Auth`.

Reply:
I think we could try to re-auth:

dialer := AuthDialer{
    Dialer: NetDialer{
    },
}

Reply:
New tests added.